### ISSUE TITLE
CRIU in centos failed

### ISSUE THREAD

**Issue by ameytotawar** [checkpoint-restore/criu#559]
Docker version - 17.03.2-ce
CRIU version - 3.5

Created MySQL DB container checkpoint without any problem.
Restoring the same MySQL DB container gives following error -
(00.314199)pie: 1: vdso: DT_SYMTAB: 0x1a8
(00.314201)pie: 1: vdso: DT_STRSZ: 0x5e
(00.314203)pie: 1: vdso: DT_SYMENT: 0x18
(00.314206)pie: 1: vdso: nbucket 0x3 nchain 0xb bucket 0x7ffe593da128 chain 0>
(00.314206)pie: 1: x7ffe593da134
(00.314212)pie: 1: vdso: image [vdso] 0x7ffe593da000-0x7ffe593dc000 [vvar] 0x>
(00.314212)pie: 1: 7ffe593d7000-0x7ffe593da000
(00.314216)pie: 1: vdso: Runtime vdso/vvar matches dumpee, remap inplace
(00.314231)pie: 1: vdso: Remap rt-vdso 0x116000 -> 0x7ffe593da000
(00.314244)pie: 1: vdso: Remap rt-vvar 0x113000 -> 0x7ffe593d7000
(00.314426)pie: 1: Error (criu/pie/restorer.c:666): wrong aio: tail=0x0 head=>
(00.314426)pie: 1: 0x0 req=0x17d old_nr=0x17f new_nr=0x2ff expect=0x17f
(00.314440)pie: 1: Error (criu/pie/restorer.c:1689): Restorer fail 1
(00.376065) mnt: Switching to new ns to clean ghosts
(00.376156) Unlink remap tmp/ibahlMdg.cr.1.ghost
(00.376235) Unlink remap tmp/ib2J6q8e.cr.2.ghost
(00.376279) Unlink remap tmp/ibECo62d.cr.3.ghost
(00.376329) Unlink remap tmp/ibISTlUb.cr.4.ghost
(00.376359) Unlink remap tmp/ib2dR32a.cr.5.ghost
(00.389414) Error (criu/cr-restore.c:2152): Restoring FAILED.

Can anyone help me in resolving this issue?

**Comment by adrianreber** [comment]
Which kernel and which operating system are you using?

**Comment by ameytotawar** [comment]
I am using below kernel and OS -
Operating System: CentOS Linux 7 (Core)
CPE OS Name: cpe:/o:centos:centos:7
Kernel: Linux 4.12.4-1.el7.elrepo.x86_64
Architecture: x86-64

**Comment by ameytotawar** [comment]
That is the server on which I need to work on.
Where can I get non CentOS CRIU?

**Comment by adrianreber** [comment]
https://criu.org/Installation

**Comment by ameytotawar** [comment]
So, do you mean to install CRIU from the package provided on criu.org website instead of installing the default package provided by CentOS?

**Comment by adrianreber** [comment]
Yes, the CRIU in CentOS works only on CentOS. If you change CentOS, by installing a non-CentOS kernel, it does not work. Unfortunately now you have to do it manually.

Curious, why are you using the non-CentOS kernel?

**Comment by adrianreber** [comment]
The original CentOS kernel has differences in AIO, which is handled by the CentOS CRIU. If you install a non-CentOS kernel the additional patch applied on the CentOS CRIU is suddenly wrong. So I still recommend to install to original kernel if possible.

If your containers processes do not use AIO it just works. It seems that mysql is suing AIO, however.
